### PR TITLE
Fix a unicode parsing error

### PR DESCRIPTION
**Pull request by obelisk**
There was an error where having a query with a path like C:\users, osquery thought that was a unicode character `sers` after `\u`.

This adds a check for double backslashes and skips them if found.

What would happen before with the new testcase:
```
➜  osquery git:(master) ✗ ./build/darwin/osquery/osquery_tests --gtest_filter="ConversionsTests.test_unicode_unescape"
Note: Google Test filter = ConversionsTests.test_unicode_unescape
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from ConversionsTests
[ RUN      ] ConversionsTests.test_unicode_unescape
WARNING: Logging before InitGoogleLogging() is written to STDERR
W0421 11:44:52.044927 2431210368 conversions.h:212] Unescaping a string with length: 28 failed at: 3
/Users/obelisk/Documents/git_repos/osquery/osquery/core/tests/conversions_tests.cpp:74: Failure
      Expected: unescapeUnicode(test.first)
      Which is: ""
To be equal to: test.second
      Which is: "c:\\\\users\\\\obelisk\\\\file.txt"
[  FAILED  ] ConversionsTests.test_unicode_unescape (1 ms)
[----------] 1 test from ConversionsTests (1 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (1 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ConversionsTests.test_unicode_unescape

 1 FAILED TEST
```

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit 48c6055aa35fc6ac7e4f81082c49226e3ac77765 (Job results: [1348](https://jenkins.osquery.io/job/osqueryPullRequestFreeBSD/1348/)) failed one or more tests (**FreeBSD**).

**Comment by osqueryer** [comment]
:no_good: The commit 87b3402b1f9265f51193ef1ae8c3c9082aa3e980 (Job results: [3393](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3393/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 87b3402b1f9265f51193ef1ae8c3c9082aa3e980 (Job results: [3797](https://jenkins.osquery.io/job/osqueryPullRequestWindows/3797/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit 87b3402b1f9265f51193ef1ae8c3c9082aa3e980 (Job results: [6907](https://jenkins.osquery.io/job/osqueryPullRequestLinux/6907/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit cca927ed2ed2667bd5cb8033f4cf8db102bd94a9 (Job results: [6910](https://jenkins.osquery.io/job/osqueryPullRequestLinux/6910/)) failed one or more tests (**Linux**).

File: osquery/core/conversions.cpp [hunk 1]
```diff
@@ -231,10 +231,15 @@ Status JSON::toString(std::string& str) const {
...
+  rj::ParseResult pr = doc_.Parse(str.c_str());
+  if (!pr) {
+    std::string message{"Cannot parse JSON: "};
+    message += GetParseError_En(pr.Code());
+    message += " Offset: ";
+    message += std::to_string(pr.Offset());
+    return Status(1, message);
   }
-  return Status();
+  return Status{0};
```
Comment (muffins) [inline osquery/core/conversions.cpp]: nit: why the initialization?

Comment (obelisk) [reply osquery/core/conversions.cpp]: No reason

Comment (muffins) [reply osquery/core/conversions.cpp]: Mind changing it back for consistency with the code base? :)

**Comment by osqueryer** [comment]
:-1: The commit f375deb4f5073e8f839ecde0b4709c5c975aa954 (Job results: [6944](https://jenkins.osquery.io/job/osqueryPullRequestLinux/6944/)) failed one or more tests (**Linux**).

**Pull request merged**